### PR TITLE
[SUCE-469] Proper donation notification.

### PR DESCRIPTION
**Pull request by gradam**

### CODE REVIEW

File: client/elements/sc-donate-success.html [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+                super.ready();
+                this.loadResources(this.resolveUrl(`/localization/elements/sc-donate-success/${this.fallbackLanguage}.json`), this.fallbackLanguage, true);
+                console.log(this.donationSuccessData);
+            }
+
+            _siteLanguageChanged() {
+                this.loadResources(this.resolveUrl(`/localization/elements/sc-donate-success/${this.language}.json`), this.language, true);
+            }
+
+            _subscriptionName(subscription, localize) {
```
Comment (TrebuhD) [inline client/elements/sc-donate-success.html]: function name is missing verb

File: client/elements/sc-donate-success.html [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+                      type: Object,
+                      statePath: 'donationSuccessData',
+                    }
+                }
+            }
+
+            ready() {
+                super.ready();
+                this.loadResources(this.resolveUrl(`/localization/elements/sc-donate-success/${this.fallbackLanguage}.json`), this.fallbackLanguage, true);
+                console.log(this.donationSuccessData);
```
Comment (TrebuhD) [inline client/elements/sc-donate-success.html]: console log

File: client/elements/sc-donate-success.html [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+            </section>
+        </div>
+    </template>
+
+    <script>
+        /**
+         * @extends {Polymer.Element}
+         * @appliesMixin Polymer.AppLocalizeBehavior
+         */
+        class SCDonateSuccess extends Polymer.mixinBehaviors([Polymer.AppLocalizeBehavior], ReduxMixin(Polymer.Element)) {
```
Comment (TrebuhD) [inline client/elements/sc-donate-success.html]: this class can `extend SCStaticPage` and drop `language` and `fallbackLanguage`

File: client/localization/elements/sc-donate-success/pl.json [hunk 4]
```diff
@@ -0,0 +1,10 @@
+{
+    "pl": {
+        "DonationOf": "Dotacja wysokości",
```
Comment (TrebuhD) [inline client/localization/elements/sc-donate-success/pl.json]: w wysokości

File: client/redux-store.html [hunk 5]
```diff
@@ -127,5 +131,18 @@
...
+    function routeChanged(state, action) {
+        let n = {
+            currentRoute: action.route,
+            previousRouteName: state.currentRoute.name
+        };
+
+        if (state.currentRoute.name === 'DONATION-SUCCESS' || state.currentRoute.name !== 'DONATE-NOW') {
+            n.donationSuccessData = {};
+        }
+        console.log(n);
```
Comment (TrebuhD) [inline client/redux-store.html]: console log

File: client/redux-store.html [hunk 6]
```diff
@@ -127,5 +131,18 @@
...
     }
 
+    function routeChanged(state, action) {
+        let n = {
+            currentRoute: action.route,
+            previousRouteName: state.currentRoute.name
+        };
+
+        if (state.currentRoute.name === 'DONATION-SUCCESS' || state.currentRoute.name !== 'DONATE-NOW') {
+            n.donationSuccessData = {};
```
Comment (TrebuhD) [inline client/redux-store.html]: if the point is to clear `donationSuccessData` when navigating away from the `DONATION-SUCCESS` page, all we need to do is dispatch a `CLEAR-DONATION-SUCCESS-DATA` action inside `detached()` in `sc-donate-success.html`.

I'd prefer not to go down this route here.

File: server/server/api/views.py [hunk 7]
```diff
@@ -737,7 +739,7 @@ def post(self):
         """
         data = json.loads(list(request.form.keys())[0])
         currency = data.get('currency')
-        amount = data.get('amount')
+        inputed_amount = data.get('amount')
```
Comment (TrebuhD) [inline server/server/api/views.py]: should be "inputted"

File: client/redux-store.html (hunk 6 above)
Comment (gradam) [reply client/redux-store.html]: But what if we close the browser when we are on donation-success page. the `CLEAR-DONATION-SUCCESS-DATA` will not be dispatched

Comment (TrebuhD) [reply client/redux-store.html]: take a look on line 125 in `redux-store.html`. You can reset the state there when loading it from `localStorage`.

File: client/redux-store.html [hunk 8]
```diff
@@ -37,15 +37,16 @@
...
+        donationSuccessData: {}
     };
 
     // The reducer accepts the current state and an action and returns a new state object
     // (note that the old state object is unmodified).
     const reducer = (state, action) => {
         switch (action.type) {
             case 'CHANGE_ROUTE':
-                return Object.assign({}, state, { currentRoute: action.route });
+                return Object.assign({}, state, { currentRoute: action.route, });
```
Comment (TrebuhD) [inline client/redux-store.html]: should the dangling `,` be here?

File: client/elements/sc-donate-success.html [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+                      statePath: 'donationSuccessData',
+                    },
+                    localizedStringPath: {
+                      type: String,
+                      value: '/localization/elements/sc-donate-success/'
+                  }
+                }
+            }
+
+        static get actions() {
```
Comment (TrebuhD) [inline client/elements/sc-donate-success.html]: indented wrong

File: server/server/search/view.py [hunk 10]
```diff
@@ -103,7 +103,7 @@ def get(self):
                         'highlight': entry['highlight'],
                         'url': f'/{uid}/{lang}/{author_uid}'
                     })
-                
+                v
```
Comment (TrebuhD) [inline server/server/search/view.py]: what's this?

File: server/server/api/views.py [hunk 11]
```diff
@@ -728,6 +729,16 @@ def get(self, word=None):
...
 class Donations(Resource):
+    def get(self):
+        data = {
+            'name': "Kuba",
+            'amount': 42.02,
+            'currency': "USD",
+            'dateTime': datetime.datetime.now().strftime('%d-%m-%y %H:%M'),
+            'subscription': False
+        }
+        self.send_email(data, 'kuba.semik@gmail.com')
```
Comment (TrebuhD) [inline server/server/api/views.py]: are we sending it to your address?

File: server/server/api/views.py [hunk 12]
```diff
@@ -728,6 +729,16 @@ def get(self, word=None):
 
 
 class Donations(Resource):
+    def get(self):
+        data = {
+            'name': "Kuba",
+            'amount': 42.02,
```
Comment (TrebuhD) [inline server/server/api/views.py]: hardcoded amount?

File: server/server/api/views.py [hunk 13]
```diff
@@ -827,6 +841,28 @@ def _get_plan(amount, currency):
                 id=plan_id)
         return plan
 
+    @staticmethod
+    @in_thread
+    def send_async_mail(msg):
+        from app import app
+        with app.app_context():
+            print('ATTEMPTING TO SEND AN EMAIL')
```
Comment (TrebuhD) [inline server/server/api/views.py]: maybe we should we log the email address in the server logs?

File: client/elements/sc-donate-success.html [hunk 14]
```diff
@@ -0,0 +1,84 @@
+<link rel="import" href="../bower_components/polymer/polymer.html">
+<link rel="import" href="../bower_components/app-localize-behavior/app-localize-behavior.html">
+<link rel="import" href="../bower_components/iron-location/iron-location.html">
+<link rel="import" href="../bower_components/iron-meta/iron-meta.html">
+<link  rel="import" href="./addons/sc-static-page.html"/>
```
Comment (TrebuhD) [inline client/elements/sc-donate-success.html]: space before rel

**Pull request merged**